pkg/build/strategies/sti/sti.go

```diff
@@ -253,7 +260,7 @@ func (b *STI) Determine(request *api.Request) (err error) {
 	// we're assuming save-artifacts to exists for embedded scripts (if not we'll
 	// warn a user upon container failure and proceed with clean build)
 	// for external save-artifacts - check its existence
-	saveArtifactsExists := request.ExternalOptionalScripts ||
+	saveArtifactsExists := !b.externalScripts[api.SaveArtifacts] ||
```

Comment:
Since I reworked the installer, the return information from installer precisely states if files was downloaded or not and the origin of the file. Since I'll be reworking this code as part of #130 I'll think about simplifying this, because I don't like the complexity of this part.